### PR TITLE
Update test defs

### PR DESCRIPTION
**Pull request by anandkumarpatel**
updated test defs

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -37,15 +37,61 @@ There are three types of tests we should be concerned about.
 
 An important note: unit tests should _never_ touch an external service (e.g. database).
 
-**Integration** tests make sure that one or more components function correctly together. This can be multiple functions within a module, multiple modules interacting with each other, or a module interacting with a database or external service.
+**Functional** tests should align with product ideas. If a user goes through a given flow, they should finish with some outcome.
+An important note: Functional tests should _never_ touch an external service (e.g. database).
```
Comment (bkendall) [inline README.md:41]: no! this is _wrong_! Functional tests should be testing end to end!

File: README.md [hunk 2]
```diff
@@ -37,15 +37,61 @@ There are three types of tests we should be concerned about.
...
 
-**Functional** tests should align with product ideas. If a user goes through a given flow, they should finish with some outcome. We've found these tests to be pretty difficult to work with as of late, and are trying to get rid of some of these.
+**Integration** tests make sure that componentes that talk to external services function correctly. These should not have any mocks and use real external deps (ex. aws, docker, githib) with fake / test tokens.
```
Comment (bkendall) [inline README.md:43]: Are you swapping the definitions of _Integration_ and _Functional_ tests?

**Comment by bkendall** [comment]
Thanks, @mention-bot! I'll take a look! :trollface:

**Comment by podviaznikov** [comment]
@bkendall haha, good input

**Comment by bkendall** [comment]
:heart:

**Pull request closed without merging**